### PR TITLE
feat: add usePreferredLanguage hook

### PR DESCRIPTION
**Pull request by nik-rev**
This hook will return the language that the user uses

### CODE REVIEW

File: src/hooks/usePreferredLanguage.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+const getPreferredLanguageSnapshot = () => {
+  return navigator.language;
+};
+
+const getPreferredLanguageServerSnapshot = () => {
+  throw Error("usePreferredLanguage is a client-only hook");
+};
+
+export default function usePreferredLanguage() {
```
Comment (maximilienGilet) [inline src/hooks/usePreferredLanguage.ts:20]: issue: Could you add the JSDoc (with types) please ?

File: src/examples/UsePreferredLanguageExample.tsx [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import usePreferredLanguage from "../hooks/usePreferredLanguage";
+
+export default function App() {
```
Comment (maximilienGilet) [inline src/examples/UsePreferredLanguageExample.tsx]: polish: could you rename this `UsePreferredLanguageExample` please ?

Comment (nik-rev) [reply src/examples/UsePreferredLanguageExample.tsx]: Oops, got it

File: src/hooks/usePreferredLanguage.ts (hunk 1 above)
Comment (nik-rev) [reply src/hooks/usePreferredLanguage.ts:20]: Yeah, how does this sound:
```
/**
 * React hook to get the user's preferred language
 * @returns {string} The user's preferred language
 */
```

**Review by maximilienGilet — APPROVED** [review]

**Pull request merged**